modules/objdetect/src/cascadedetect.cpp

```diff
@@ -587,7 +587,7 @@ bool HaarEvaluator::read(const FileNode& node, Size _origWinSize)
     localSize = lbufSize = Size(0, 0);
     if (ocl::haveOpenCL())
     {
-        if (ocl::Device::getDefault().isAMD())
+        if (ocl::Device::getDefault().isAMD() || ocl::Device::getDefault().isIntel())
```

Comment:
My current test platform is Baytail (Atom), OpenCL branch (based on Beignet) on Linux is about faster than 35% than CPU branch after enabling this patch. If no this patch, it will cause GPU hang and failed on OCL_Cascade_Image_MinSize_CascadeClassifier.CascadeClassifier/*. So I think this issue may cause OpenCL branch is slower than CPU branch. If possible, could you please test again with this patch?